test/e2e/branding_test.go

```diff
@@ -0,0 +1,192 @@
+		t.Fatalf("could not get console-config configmap, %v", err)
+	}
+
+	return client, originalConfigBrand, originalManagedConfigMap.Data
+}
+
+func cleanupBrandingTestCase(t *testing.T, client *testframework.Clientset, originalConfigBrand operatorsv1.Brand, originalManagedConfigMapData map[string]string) {
+
+	setOperatorConfigBrand(t, client, originalConfigBrand)
+
+	managedConfigMap := generateTestConfigMap(operatorsv1.BrandOKD)
+	if originalManagedConfigMapData != nil {
+		managedConfigMap.Data = originalManagedConfigMapData
+	}
+	_, err := client.ConfigMaps(consoleapi.OpenShiftConfigManagedNamespace).Update(managedConfigMap)
+	if err != nil {
+		t.Fatalf("could not reset managed config map  %v", err)
+	}
+
+}
+
+// TestOperatorConfigBranding() tests that changing the brand value on the operator-config
+// will result in the brand being set on the console-config in openshift-console.
+// Implicitly it ensures that the operator-config brand overrides brand set on
+// console-config in openshift-config-managed, if the managed configmap exists.
+func TestOperatorConfigBranding(t *testing.T) {
+	client, originalConfigBrand, originalManagedConfigMapData := setupBrandingTestCase(t)
+	defer cleanupBrandingTestCase(t, client, originalConfigBrand, originalManagedConfigMapData)
+	// Set a temporary managed config to test it does not override the operator config values
+	_, err := updateOrCreateConsoleConfigMap(client, generateTestConfigMap(operatorsv1.BrandOKD))
+	if err != nil {
+		t.Fatalf("error: could not apply managed config map %v", err)
+	}
+
+	for _, expectedBrand := range validOperatorConfigBrands {
+		t.Logf("update operator with %v", expectedBrand)
+		// helper to update the operator config
+		setOperatorConfigBrand(t, client, expectedBrand)
+		// now check if it has set the brand
+		err = wait.Poll(1*time.Second, pollTimeout, func() (stop bool, err error) {
+			gotBrand := getConsoleBrand(t, client)
+			return (gotBrand == expectedBrand), nil
+		})
+	}
+}
+
+// Test setting brand via the config map in the openshift-config-managed namespace, this requires the operator config not be set
+func TestBrandingFromManagedConfigMap(t *testing.T) {
+	client, originalConfigBrand, originalManagedConfigMapData := setupBrandingTestCase(t)
+	defer cleanupBrandingTestCase(t, client, originalConfigBrand, originalManagedConfigMapData)
+	// Set operator config to empty so it does not override the managed config map values
+	setOperatorConfigBrand(t, client, "")
+
+	for _, expectedBrand := range validManagedConfigMapBrands {
+		t.Logf("update data for the config map in openshift-config-managed namespace with %v", expectedBrand)
+		// Create new configmap for test
+		_, err := updateOrCreateConsoleConfigMap(client, generateTestConfigMap(expectedBrand))
+		if err != nil {
+			t.Fatalf("error: could not apply managed config map %v", err)
+		}
+
+		err = wait.Poll(1*time.Second, pollTimeout, func() (stop bool, err error) {
```

Comment:
Same as above, your `t.Fatalf()` is gone, I don't think your test is going to report anything if it fails.

Reply:
fixed